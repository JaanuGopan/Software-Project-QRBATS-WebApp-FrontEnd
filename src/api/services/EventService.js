import axios from "axios";
import ApiConstants from "../config/ApiConstants";

class EventService {
  static async fetchEvents() {
    try {
      const response = await axios.post(ApiConstants.getAllEventUrl);
      return response.data;
    } catch (error) {
      console.error("Error fetching events:", error);
      throw error; // Re-throw the error to handle it in the component
    }
  }

  static async saveEvent(
    eventName,
    eventDate,
    eventValidDate,
    eventTime,
    eventEndTime,
    eventVenue,
    eventRole,
    moduleName,
    userId
  ) {
    try {
      const response = await axios.post(ApiConstants.createEventUrl, {
        eventName: eventName,
        eventDate: eventDate,
        eventValidDate: eventValidDate,
        eventTime: eventTime,
        eventEndTime: eventEndTime,
        eventVenue: eventVenue,
        eventRole: eventRole,
        eventModuleName: moduleName,
        eventAssignedUserId: userId,
      });
      return response.data;
    } catch (error) {
      console.error("Event creation failed", error);
    }
  }
}

export default EventService;
