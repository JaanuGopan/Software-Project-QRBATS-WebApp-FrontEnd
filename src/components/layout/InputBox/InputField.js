import React from "react";
import "./inputfield.css";

const InputField = ({ placeholder, value, onChange, inputType }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className="input-field"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default InputField;
