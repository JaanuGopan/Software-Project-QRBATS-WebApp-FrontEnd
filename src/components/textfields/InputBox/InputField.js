import React from "react";
import "../../../pages/Event/EventCreation/EventCreation.css";

const InputField = ({ placeholder, value, onChange, inputType }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className="form-control mb-2"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default InputField;
