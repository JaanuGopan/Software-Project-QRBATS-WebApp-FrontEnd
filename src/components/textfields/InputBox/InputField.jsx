import React from 'react';
import '../../../pages/Event/EventCreation/EventCreation.css';

const InputField = ({ placeholder, value, onChange, inputType, error, helperText }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className={`form-control mb-2 ${error ? 'is-invalid' : ''}`}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
      {error && <div className="invalid-feedback">{helperText}</div>}
    </div>
  );
};

export default InputField;
