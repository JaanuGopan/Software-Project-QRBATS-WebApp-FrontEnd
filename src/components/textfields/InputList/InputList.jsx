import React from 'react';
import '../../../pages/Event/EventCreation/EventCreation.css';

const InputList = ({ placeholder, value, onChange, inputType, list, initialValue }) => {
  return (
    <div className="input-with-icon">
      <select
        required
        value={value}
        onChange={onChange}
        placeholder={placeholder}
        className="form-control mb-2"
      >
        {initialValue ? null : <option value="">{initialValue}</option>}
        {[initialValue, ...list.filter((option) => option !== initialValue)].map(
          (option, index) => (
            <option key={index} value={option} selected={option === initialValue}>
              {option}
            </option>
          )
        )}
      </select>
    </div>
  );
};

export default InputList;
