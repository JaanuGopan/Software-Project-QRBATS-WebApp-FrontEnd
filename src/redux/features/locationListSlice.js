import { createSlice } from "@reduxjs/toolkit";

export const locationListSlice = createSlice({
  name: "locationList",
  initialState: ["NCC", "LT1", "LT2", "Auditorium", "DEIE", "DMME", "DCEE"],
  reducers: {
    setLocationList: (state, action) => {
      state = [];
      state.push(...state, action.payload);
    },
    resetLocationList: (state) => {
      state = [];
    },
  },
});

export const { setLocationList, resetLocationList } = locationListSlice.actions;
export default locationListSlice.reducer;
