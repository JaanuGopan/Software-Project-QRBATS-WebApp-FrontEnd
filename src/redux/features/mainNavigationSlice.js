import { createSlice } from "@reduxjs/toolkit";

export const mainNavigationSlice = createSlice({
  name: "mainNavigation",
  initialState: {
    sideBarIndex:
      localStorage.getItem("sideBarIndex") != null
        ? parseInt(localStorage.getItem("sideBarIndex"))
        : 0,
  },
  reducers: {
    setSideBarIndex: (state, action) => {
      state.sideBarIndex = action.payload;
      localStorage.setItem("sideBarIndex", action.payload);
    },
    resetSideBarIndex: (state) => {
      state.mainNavigation = 0;
      localStorage.removeItem("sideBarIndex");
    },
  },
});

export const { setSideBarIndex, resetSideBarIndex } =
  mainNavigationSlice.actions;
export default mainNavigationSlice.reducer;
