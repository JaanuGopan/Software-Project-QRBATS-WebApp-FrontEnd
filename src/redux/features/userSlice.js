import { createSlice } from '@reduxjs/toolkit';

export const userSlice = createSlice({
  name: 'user',
  initialState: {
    user: JSON.parse(localStorage.getItem('user')) || null,
    isLoggedIn: !!localStorage.getItem('authToken'),
    authToken: localStorage.getItem('authToken') || null,
  },
  reducers: {
    login: (state, action) => {
      state.user = action.payload;
    },
    logout: (state) => {
      state.user = null;
      localStorage.removeItem('user');
      localStorage.removeItem('authToken');
    },
  },
});

export const { login, logout } = userSlice.actions;
export const selectUser = (state) => state.user.user;
export default userSlice.reducer;
export const selectIsLoggedIn = (state) => state.user.isLoggedIn;
