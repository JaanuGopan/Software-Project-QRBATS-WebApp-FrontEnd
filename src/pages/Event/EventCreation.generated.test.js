import renderer from 'react-test-renderer';

import eventCreationImage from "../../assets/Images/signin/Signin.jpeg";
import React, { useState, useRef } from "react";
import { IoMdCloseCircleOutline } from "react-icons/io";
import toast, { Toaster } from "react-hot-toast";
import { useSelector } from "react-redux";
import { selectUser } from "../../redux/features/userSlice";
import QRCode from "qrcode.react";
import InputList from "../../components/textfields/InputList/InputList";
import InputField from "../../components/textfields/InputBox/InputField";
import EventService from "../../api/services/EventService";
import EventCreationComponent from "./EventCreation";

jest.mock("./EventCreation/EventCreation.css");
jest.mock("../../assets/Images/signin/Signin.jpeg");
jest.mock("react-icons/io");
jest.mock("react-hot-toast");
jest.mock("react-redux");
jest.mock("../../redux/features/userSlice");
jest.mock("qrcode.react");
jest.mock("../../components/textfields/InputList/InputList");
jest.mock("../../components/textfields/InputBox/InputField");
jest.mock("../../api/services/EventService");

const renderTree = tree => renderer.create(tree);
describe('<EventCreationComponent>', () => {
  it('should render component', () => {
    expect(renderTree(<EventCreationComponent 
    />).toJSON()).toMatchSnapshot();
  });
  
});