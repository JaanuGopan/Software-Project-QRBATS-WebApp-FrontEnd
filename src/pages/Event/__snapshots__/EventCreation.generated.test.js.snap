// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventCreationComponent> should render component 1`] = `
<div
  className="event-main-container2"
>
  <h2>
    Create Event
  </h2>
  <div
    className="eventCreation-field"
  >
    <img
      alt="Logo"
      className="Create-logo"
      src="Signin.jpeg"
    />
    <div
      className="eventCreation-input-field"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="input-with-icon"
        />
        <div
          className="date-div"
        >
          <div
            className="eventCreation-form"
          >
            <label
              className="date-label"
              htmlFor="eventDate"
            >
              Event Starting Date
            </label>
          </div>
          <div
            className="eventCreation-form"
          >
            <label
              className="date-label"
              htmlFor="eventDate"
            >
              Event Ending Date
            </label>
          </div>
        </div>
        <div
          className="eventCreation-form"
        >
          <label
            className="date-label"
            htmlFor="eventDate"
          >
            Venue
          </label>
        </div>
        <div
          className="eventCreation-form"
        >
          <label
            className="form-label"
            htmlFor="eventDate"
          >
            Event Starting Time
          </label>
          <div
            className="eventCreation-form"
          >
            <label
              className="form-label"
              htmlFor="enevtEndTime"
            >
              Event Ending Time
            </label>
          </div>
        </div>
        <div
          className="eventCreation-form"
        />
        <div
          className="d-flex justify-content-between mr-3 mt-3"
        >
          <button
            className="btn btn-primary w-90"
            onClick={[Function]}
            type="submit"
          >
            Create Event
          </button>
          <button
            className="btn btn-danger w-30"
            onClick={[Function]}
          >
            Clear
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
